packages/nodes-from-anchor/src/v01/typeNodes/StructFieldTypeNode.ts

```diff
@@ -0,0 +1,23 @@
+import { KINOBI_ERROR__UNRECOGNIZED_ANCHOR_IDL_TYPE, KinobiError } from '@kinobi-so/errors';
+import { StructFieldTypeNode, structFieldTypeNode } from '@kinobi-so/nodes';
+
+import { IdlV01Field, IdlV01Type } from '../idl';
+import { typeNodeFromAnchorV01 } from './TypeNode';
+
+export function structFieldTypeNodeFromAnchorV01(idl: IdlV01Field | IdlV01Type): StructFieldTypeNode {
+    const field = idl as IdlV01Field;
+
+    const isFieldDefinition = 'name' in field && 'type' in field;
+
+    if (isFieldDefinition) {
+        return structFieldTypeNode({
+            docs: field.docs ?? [],
+            name: field.name,
+            type: typeNodeFromAnchorV01(field.type),
+        });
+    }
+
+    throw new KinobiError(KINOBI_ERROR__UNRECOGNIZED_ANCHOR_IDL_TYPE, {
+        idlType: JSON.stringify(idl),
+    });
```

Comment:
Not accounting for the fact that struct objects can have field that are leaf types (so name associated).

```
{
 kind: "struct",
 fields: ["u32", "pubkey"]
}
```

Reply:
Since you're already tackling this use-case in the `typeNodeFromAnchorV01` function, I think it's fine to throw here. I would even only accept the `IdlV01Field` type as argument here and fail higher in the `structTypeNodeFromAnchorV01` helper instead.

I'll comment on the relevant bit of code but if you manage to capture the case were all fields are unnamed for tuples and all fields are named for structs, then we're only throwing if there is a mix and match of named and unnamed fields which I don't think we should aim to support.

EDIT: In fact, based on that IDL type, a mix and match should never occur:

```ts
export type IdlV01DefinedFields = IdlV01DefinedFieldsNamed | IdlV01DefinedFieldsTuple;
export type IdlV01DefinedFieldsNamed = IdlV01Field[];
export type IdlV01DefinedFieldsTuple = IdlV01Type[];
```